app/modules/intelligence/tools/web_tools/web_search_tool.py

```
@@ -87,11 +89,19 @@ async def _make_llm_call(self, query: str) -> Dict[str, Any]:
 
-            return {
+            result = {
                 "success": True,
                 "content": text_response or "",
                 "citations": [],
             }
+
+            # Truncate response if it exceeds character limits
+            truncated_result = truncate_dict_response(result)
+            if len(str(result)) > 80000:
+                logging.warning(
```

Ruff (0.14.10)

Undefined name `logging`

(F821)